lib/child_process.js

```diff
@@ -137,7 +138,8 @@ function normalizeExecArgs(command, options, callback) {
 
   // Make a shallow copy so we don't clobber the user's options object.
   options = { ...options };
-  options.shell = typeof options.shell === 'string' ? options.shell : true;
+  if (typeof options.shell !== 'string' && !Array.isArray(options.shell))
+    options.shell = true;
```

Comment:
can you pls explain this change?

Reply:
otherwise when `options.shell` is `['xx', 'xxx']`, it will be changed to `true`